packages/lib/src/vis/heatmap/utils.ts

```diff
@@ -127,15 +101,15 @@ export function getLinearGradient(
 }
 
 export function getAxisValues(
-  rawValues: number[] | undefined,
+  rawValues: NumArray | undefined,
   pixelCount: number
 ): number[] {
   if (!rawValues) {
     return range(pixelCount + 1);
   }
 
   if (rawValues.length === pixelCount + 1) {
-    return rawValues;
+    return isTypedArray(rawValues) ? [...rawValues] : rawValues;
```

Comment:
Even though axis values are now fetched as typed arrays, I still convert them to `number[]` here. Turns out it was a bit complicated to handle typed array in `getValueToIndexScale`. Maybe I'll revisit this later on.